docker-compose.yml

```diff
@@ -903,35 +902,6 @@ services:
       /bin/bash -c "
         /arrow/ci/scripts/r_sanitize.sh /arrow"
 
-  ################################ Rust #######################################
-
-  debian-rust:
```

Comment:
Even if we plan to use plain GHA jobs without docker-compose it'd be nice to keep it around to let contributors execute rust builds without installing the toolchain locally.

Reply:
The arguments I used to remove it:

* this compose and associated dockerfile are no longer tested under the CI, and their primary purpose was to be used in the CI
* the dockerfile is not needed, as the default rust image is sufficient
* the compose contains non-trivial aspects, such as:
   * volume mounting with `delegate`
   * using the no-longer needed `rust_test.sh`
   * non-default `CARGO_*` env variables
* `docker-compose run debian-rust` is IMO less transparent to a user about what is happening, and the person needs to go to a +1k LOC file to understand.

In opposition, this whole project can now be built using

```bash
docker run --rm -v $(pwd)/rust:/rust -it rust /bin/bash -c "cd /rust && cargo build"
```

which IMO is really simple to understand and reproduce. Wouldn't this command on the README be sufficient for contributors?

In the end, I evaluated that having these around would be more confusing than just having the user to follow standard rust practices of building a rust project (including using the official docker image), and therefore there was no need for the maintenance burden of having these around.

Reply:
Okay, sounds good to me - lucky us that we have cargo around. 

Please document how to reproduce the builds locally with docker.

Reply:
I have added a section on the README on how to compile, which includes both pure `cargo` and `docker` variants.